src/coreclr/jit/importercalls.cpp

```diff
@@ -2865,35 +2867,59 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 // to `true` or `false`
                 // e.g., `typeof(int).IsValueType` => `true`
                 // e.g., `typeof(Span<int>).IsByRefLike` => `true`
-                if (impStackTop().val->IsCall())
+                CORINFO_CLASS_HANDLE hClass = NO_CLASS_HANDLE;
+                if (gtIsTypeof(impStackTop().val, &hClass))
                 {
-                    GenTreeCall* call = impStackTop().val->AsCall();
-                    if (call->gtCallMethHnd == eeFindHelper(CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPE))
+                    switch (ni)
                     {
-                        assert(call->gtArgs.CountArgs() == 1);
-                        CORINFO_CLASS_HANDLE hClass =
-                            gtGetHelperArgClassHandle(call->gtArgs.GetArgByIndex(0)->GetEarlyNode());
-                        if (hClass != NO_CLASS_HANDLE)
+                        case NI_System_Type_get_IsEnum:
+                        case NI_System_RuntimeType_get_IsActualEnum:
                         {
-                            switch (ni)
+                            CORINFO_CLASS_HANDLE hClassUnderlying = NO_CLASS_HANDLE;
+                            TypeCompareState state = info.compCompHnd->isEnum(hClass, &hClassUnderlying);
+                            if (state == TypeCompareState::May ||
+                               (state == TypeCompareState::Must && hClassUnderlying == NO_CLASS_HANDLE))
```

Comment:
```suggestion
                            TypeCompareState state = info.compCompHnd->isEnum(hClass, nullptr);
                            if (state == TypeCompareState::May)
```
The handle is not needed here at all.

Reply:
I think that the NAOT implementation could theoretically return `Must` with a null underlying type for generics.